burstphoto/denoise.swift

```diff
@@ -319,32 +318,36 @@ func calculate_temporal_average(progress: ProcessingProgress, mosaic_pattern_wid
         }
     }
     
-    if (!uniform_exposure && white_level != -1 && black_level[0][0] != -1 && mosaic_pattern_width == 2) {
+    // if color_factor is NOT available, it is set to a negative value earlier in the pipeline
+    if (white_level != -1 && black_level[0][0] != -1 && color_factors[0][0] > 0 && mosaic_pattern_width == 2) {
         
```

Comment:
```suggestion
        // temporal averaging with extrapolation of highlights or exposure weighting
        
        // The averaging is performed in two steps:
        // 1. Add textures
        // 2. Divide the resulting summed texture by a per-pixel norm to get the final values
        
        // The per-pixel norm has two contributions: norm(x, y) = norm_texture(x, y) + norm_scalar
        // 1. A per-pixel calculated by add_texture_exposure which accounts for the different weight given to each pixel based on its brightness
        // 2. A global scalar if a given image has the same exposure as the reference reference exposure (the darkest one).
        //
        // For 2, a single scalar is used, rather than adding to the norm texture since it's a uniform value and doing so will save a lot of additions.
```

Reply:
Added a description very similar to your suggestion.